controller/modelmanager/createmodel.go

```diff
@@ -69,6 +70,27 @@ func (c ModelConfigCreator) NewModelConfig(
 		attrs[config.UUIDKey] = uuid.String()
 	}
 	attrs[config.TypeKey] = cloud.Type
+
+	// We need to get the system-identity public key to be added to the
```

Comment:
We should have a much better way of identifying the "system-identity" public key for the controller than parsing its model-config.
Do we not track that anywhere else?

Another possible fix (better? roughly the same, I think), is to ensure that system-identity gets injected into model-defaults whenever that key gets touched. I think its at least worth evaluating if that feels better.

Certainly the longer term fix is to properly model what identities are allowed to ssh, and then populate authorized-keys from that rather than just a string that we parse in, mutate, and then spit out.

Reply:
Agree with John here. Can we ensure model defaults is correctly maintained?

Reply:
When you say model defaults, are you talking about the user-manageable `juju model-defaults` as in https://juju.is/docs/juju/juju-model-defaults? If so, I don't think it's a good idea to set the system-identity key in there, since it's something juju needs to function and the user shouldn't even know about its existence.

Reply:
I'm trying to pass it as `ControllerInheritedConfig` at bootstrap, let's see where that brings me